presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java

```diff
@@ -0,0 +1,425 @@
+     * @param rowIDRanges Collection of Ranges based on any predicate from the row ID. Used to drop and row
+     * IDs that are not in one of these Ranges
+     * @return A collection of Ranges containing row IDs in the main table to scan
+     * @throws TableNotFoundException
+     */
+    private List<Range> getIndexRanges(String indexTable,
+            Map<AccumuloColumnConstraint, Collection<Range>> constraintRangePairs,
+            Collection<Range> rowIDRanges)
+            throws TableNotFoundException
+    {
+        Set<Range> finalRanges = null;
+        // For each column/constraint pair
+        for (Entry<AccumuloColumnConstraint, Collection<Range>> e : constraintRangePairs
+                .entrySet()) {
+            // Create a batch scanner against the index table, setting the ranges
+            BatchScanner scan = conn.createBatchScanner(indexTable, auths, 10);
+            scan.setRanges(e.getValue());
+
+            // Fetch the column family for this specific column
+            Text cf = new Text(Indexer.getIndexColumnFamily(e.getKey().getFamily().getBytes(),
+                    e.getKey().getQualifier().getBytes()).array());
+            scan.fetchColumnFamily(cf);
+
+            // For each entry in the scanner
+            Set<Range> columnRanges = new HashSet<>();
+            for (Entry<Key, Value> entry : scan) {
+                entry.getKey().getColumnQualifier(tmpCQ);
+
+                // Add to our column ranges if it is in one of the row ID ranges
+                if (inRange(tmpCQ, rowIDRanges)) {
+                    columnRanges.add(new Range(tmpCQ));
+                }
+            }
+
+            LOG.info("Retrieved %d ranges for column %s", columnRanges.size(),
+                    e.getKey().getName());
+
+            // If finalRanges is null, we have not yet added any column ranges
+            if (finalRanges == null) {
+                finalRanges = new HashSet<>();
+                finalRanges.addAll(columnRanges);
+            }
+            else {
+                // Retain only the row IDs for this column that have already been added
+                // This is your set intersection operation!
+                finalRanges.retainAll(columnRanges);
+            }
+
+            // Close the scanner
+            scan.close();
+        }
+
+        // Return the final ranges for all constraint pairs
+        return new ArrayList<>(finalRanges);
+    }
+
+    /**
+     * Bins the given collection of ranges into metadata for the Presto splits
+     *
+     * @param numRangesPerBin Number of ranges to put in a single metadata object
+     * @param splitRanges The ranges to add
+     * @param prestoSplits The output collection for the tablet metadata
+     */
+    private void binRanges(int numRangesPerBin, List<Range> splitRanges,
+            List<TabletSplitMetadata> prestoSplits)
+    {
+        // Shuffle the handles to give an even distribution across all tablet splits
```

Comment:
Shouldn't `numRangesPerBin` be asserted to be > 0?
If so, please `checkState` / `checkArgument`.